app/code/Magento/Sales/Model/Order.php

```diff
@@ -676,13 +676,12 @@ public function canCreditmemoForZeroTotal($totalRefunded)
         //check if total paid is less than grandtotal
         $checkAmtTotalPaid = $totalPaid <= $this->getGrandTotal();
         //case when amount is due for invoice
-        $dueAmountCondition = $this->canInvoice() && ($checkAmtTotalPaid);
+        $hasDueAmount = $this->canInvoice() && ($checkAmtTotalPaid);
         //case when paid amount is refunded and order has creditmemo created
-
-        $creditmemos = ($this->getCreditmemosCollection() === false) ?
+        $creditMemos = ($this->getCreditmemosCollection() === false) ?
```

Comment:
@viral-wagento please revert changes in `app/code/Magento/Sales/Model/Order.php` related to credit memos too.

Reply:
@viral-wagento please check https://github.com/magento/magento2/pull/19288/files to see all items which are not done, apply changes and then squash them into a single commit.

Reply:
@viral-wagento almost there: https://github.com/magento/magento2/pull/19288/files

Please remove unneeded space "credit memo" -> "creditmemo" and squash changes into a single commit performing a force push.

Reply:
@viral-wagento and in comment

```
// Case when Adjustment Fee (adjustment_negative) has been used for first credit memo
```

Reply:
@orlangur , okay going to change there too.
But then what about `Retrieve order credit memo (refund) availability` `Retrieve credit memo for zero total refunded availability.` and ` Retrieve credit memo for zero total availability.`

They are function description

Reply:
@viral-wagento thanks! No need to change, as they were there before, not really worth fixing.

Changes look good to me now, please squash them into a single commit. Talk to me if you need any assistance with that.

Reply:
(Y)